Technico/Services/UserService.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Technico.Models;
+using Technico.Repositories;
+using Technico.Dtos;
+
+namespace Technico.Services;
+
+public class UserService
+{
+    private readonly UserRepository _userRepository;
+
+    public UserService(UserRepository userRepository)
+    {
+        _userRepository = userRepository;
+    }
+
+    public async Task<UserResponseDTO?> CreateAsync(UserRequestDTO createDto)
+    {
+        var users = await _userRepository.GetAllAsync();
+
+        bool VATexists = users.Any(u => u.VATNumber == createDto.VATNumber);
```

Comment:
The CreateAsync method retrieves all users via GetAllAsync to check for existing VAT and email.